llvm/lib/Transforms/Scalar/MemCpyOptimizer.cpp

```diff
@@ -1569,6 +1569,10 @@ bool MemCpyOptPass::performStackMoveOptzn(Instruction *Load, Instruction *Store,
               continue;
             }
           }
+          if (UI->hasMetadata(LLVMContext::MD_tbaa))
+            NoAliasInstrs.insert(UI);
```

Comment:
Now that this no longer holds only `!noalias`, rename to `AAMetadataInstrs` or something?

Reply:
Though at this point it would probably make sense to just collect all instructions here (regardless of whether they have metadata) and call setMetadata on all of them, so we don't have to list the MD kinds twice.

Reply:
Done, I just collect all instructions here